apps/src/studio/studio.js

```diff
@@ -1184,6 +1184,20 @@ function handleActorCollisionsWithCollidableList (
             callHandler('whenGetAllItems');
           }
 
+          if (list == Studio.items) {
+            var className = collidable.className;
+            var itemCount = 0;
+            for (var j = 0; j < list.length; j++) {
+              if (className == list[j].className) {
+                itemCount++;
+              }
+            }
+
+            if (itemCount == 1) {
+              callHandler('whenGetAll-' + className);
```

Comment:
TODO: we will need to add matching `executeQueue()` calls in order for this to work in blockly. I'll take care of that when I get to level 15 in the blockly script.